releases-implementation/README.md

```diff
@@ -0,0 +1,415 @@
+	- If there is a need to have the new app version tested and delivered ASAP, e.g. in case of CVEs, the existing/draft releases repo PR does not have to be updated, and another PR can be created specifically for the new app version (after this the previously created PR can be rebased/updated).
+- The delivery of the new app versions is not entangled with the delivery of cluster-$provider app Helm chart changes.
+- The team that owns the app can create the new release on their own and they do not need to ask other teams for the review/approval, since the unit tests in the releases repository verify that the newly created releases respects the defined release rules, e.g. new app patch delivered to all providers, new app minor must be delivered in a new minor release, etc.
+
+Changes to the current e2e tests would be minimal initially.
+- Testing of cluster-$provider app would require one addition, which would be:
+	- Looking up the latest release and setting the release label on the App CR.
+	- Like today, cluster-test-suites would use custom version of cluster-$provider app from the branch it is testing (in which case the mutating webhook would not set the cluster-$provider app version based on the Release CR).
+- The existing app testing where app versions are overridden via ephemeral Helm values would continue working as is.
+- As a future improvement, cluster-test-suites could run all tests not only for the latest release, but for all releases which use the same major cluster-$provider app version.
+
+#### 4.2.4. Multiple release models and release channels
+
+With cluster-$provider app releases, while we can use git branches to have multiple major versions, all that is only one way to create releases. We can have multiple major versions, but it’s still the same apps being versioned in the same way in every release.
+
+Since app and component versions are embedded in the cluster-$provider app, we cannot combine different versions of different apps in different ways (unless we make app and component versions configurable, which then opens the door to a whole new set of issues). We cannot have another release model where e.g. versions of Kubernetes, OS, CNI and CPI are a part of the release, and all other apps are always at their latest versions. Or a release model where we use LTS release of OS, CPI, CNI and other apps that provide a LTS release (or something similar).
+
+OTOH with the releases repository, we can easily create different directories for different release models, where we can combine the versions of apps and components in whatever way and where we can update different release models with different frequency and with different rules. Therefore it would be relatively easy to have one release model where app versions are not even part of the release, or are decoupled and continuously updated. Or another “slower” release model with long-term support where apps and components are on their LTS versions and are updated more slowly.
```

Comment:
I don't agree that this is "easy" in any way. It adds quite a bit of complexity to release management (what PRs need to be created when in what order), testing (each variation adds another suite of tests that need to be run each time) and around upgrades (can a cluster move to a different release channel? Can they then move back?)

Reply:
I would guess that handling such things in folder structures is easier to implement and gives more flexibility than branches do, although with branches you can rely on existing git tooling that does sometimes already support such features (as many projects out there do work with branches).
Maybe at least for the simple cases that we are seeing already (releasing a new WC version based on a new app release or k8s/OS release, we could already make the process clear and think of how we would automate this.

Reply:
That can however be separate from this RFC in general

Reply:
We shouldn't add more complexity/feature than what we offered with Vintage releases. It will be very hard to implement and understand. Better to bring CAPI releases on par with vintage releases, and then see how it goes.